From f2db5d09f0a5648b91ef015fe00c53090d7159fe Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Wed, 30 Aug 2023 21:15:14 +0200
Subject: [PATCH 1/9] Fix option -F in e(e)pic and latex outputs, ticket #157

With the -F option given, font family, shape or weight should not be set
in epic, eepic, latex, pdftex_t and pstex_t output formats. Instead,
with commit [a31b55] a regression was introduced wherupon -F suppressed
setting of the font size and other font properties were set. Restore the
expected behavior.
---
 fig2dev/dev/genepic.c  | 6 +++---
 fig2dev/dev/genlatex.c | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/fig2dev/dev/genepic.c b/fig2dev/dev/genepic.c
index 6c9e50d..ab398f0 100644
--- a/fig2dev/dev/genepic.c
+++ b/fig2dev/dev/genepic.c
@@ -166,7 +166,7 @@ static char *FillCommands(int style, int color);
 /* Local to the file only */
 static double	Threshold;
 static bool	linew_spec = false;
-static bool	select_fontsize = true;
+static bool	select_fontname = true;
 static int	CurWidth = 0;
 static int	LineStyle = SOLID_LINE;
 static int	LLX = 0, LLY = 0;
@@ -305,7 +305,7 @@ genepic_option(char opt, char *optarg)
 		break;
 
 	case 'F':
-		select_fontsize = false;
+		select_fontname = false;
 		break;
 
 	case 'R':
@@ -1257,7 +1257,7 @@ genepic_text(F_text *text)
 		fprintf(tfp, "\\makebox(0,0)%s{", tpos);
 	fprintf(tfp, "\\smash{");
 
-	select_font(text, select_fontsize, true, false);
+	select_font(text, true, select_fontname, false);
 	put_string(text->cstring, text->font, special_text(text),
 			need_conversion);
 
diff --git a/fig2dev/dev/genlatex.c b/fig2dev/dev/genlatex.c
index 7282efc..161e1ee 100644
--- a/fig2dev/dev/genlatex.c
+++ b/fig2dev/dev/genlatex.c
@@ -110,7 +110,7 @@ char		thickdot[] = THICKDOT;
 char		thin_ldot [] = THIN_LDOT;
 char		thick_ldot[] = THICK_LDOT;
 
-static bool	select_fontsize = true;
+static bool	select_fontname = true;
 static	int	verbose = 0;
 double		dash_mag = 1.0;
 int		thick_width = 2;
@@ -237,7 +237,7 @@ genlatex_option(char opt, char *optarg)
 		break;
 
 	case 'F':
-		select_fontsize = false;
+		select_fontname = false;
 		break;
 
 	case 'f':			/* set default text font */
@@ -888,7 +888,7 @@ genlatex_text(F_text *t)
 	/* smash is used to position text at baseline */
 	fprintf(tfp, "\\makebox(0,0)%s{\\smash{", tpos);
 
-	select_font(t, select_fontsize, true, false);
+	select_font(t, true, select_fontname, false);
 
 	set_color(t->color);
 
-- 
2.34.1

